src/test/java/uk/gov/companieshouse/limitedpartnershipsapi/service/GeneralPartnerServiceValidateTest.java

```diff
@@ -176,6 +191,43 @@ void shouldThrowResourceNotFoundExceptionWhenGeneralPartnerIdAndTransactionIdDoN
         assertThrows(ResourceNotFoundException.class, () -> service.validateGeneralPartner(transaction, GENERAL_PARTNER_ID));
     }
 
+    @Test
+    void shouldReturnErrorWhenNotDisqualifiedStatementCheckedIsNull() throws ServiceException {
+        // given
+        GeneralPartnerDao generalPartnerDao = createPersonDao();
+        generalPartnerDao.getData().setNotDisqualifiedStatementChecked(null);
+
+        when(repository.findById(generalPartnerDao.getId())).thenReturn(Optional.of(generalPartnerDao));
+
+        // when
+        List<ValidationStatusError> results = service.validateGeneralPartner(transaction, GENERAL_PARTNER_ID);
+
+        // then
+        verify(repository).findById(generalPartnerDao.getId());
+        assertThat(results)
+                .extracting(ValidationStatusError::getError, ValidationStatusError::getLocation)
+                .containsExactlyInAnyOrder(
+                        tuple("Not Disqualified Statement must be checked", GeneralPartnerDataDto.NOT_DISQUALIFIED_STATEMENT_CHECKED_FIELD));
+    }
+
+    @Test
+    void shouldNotReturnErrorWhenNotDisqualifiedStatementCheckedIsNullForATransitionFiling() throws ServiceException {
+        // given
+        GeneralPartnerDao generalPartnerDao = createPersonDao();
+        generalPartnerDao.getData().setNotDisqualifiedStatementChecked(null);
+        transaction.setFilingMode(IncorporationKind.TRANSITION.getDescription());
+
+        when(repository.findById(generalPartnerDao.getId())).thenReturn(Optional.of(generalPartnerDao));
+        when(companyProfileApi.getDateOfCreation()).thenReturn(LocalDate.of(2022, 1, 3));
+
+        // when
+        List<ValidationStatusError> results = service.validateGeneralPartner(transaction, GENERAL_PARTNER_ID);
+
+        // then
+        verify(repository).findById(generalPartnerDao.getId());
+        assertThat(results).isEmpty();
```

Comment:
Could we changes the name "results" to "errors"?

Reply:
done